xwiki-pro-macros-api/src/main/java/com/xwiki/macros/viewfile/internal/thumbnail/generators/OfficeThumbnailGenerator.java

```diff
@@ -61,46 +51,23 @@
 @Component
 @Singleton
 @Named(OfficeThumbnailGenerator.HINT)
-public class OfficeThumbnailGenerator extends AbstractOfficePdfThumbnailGenerator implements Initializable
+public class OfficeThumbnailGenerator extends AbstractOfficePdfThumbnailGenerator
 {
     /**
      * Component hint.
      */
     public static final String HINT = "office";
 
-    /**
-     * The office server configuration.
-     */
     @Inject
-    private OfficeServerConfiguration officeServerConfig;
-
-    @Inject
-    private OfficeServer officeServer;
-
-    private OfficeManager officeManager;
+    private OfficeManagerWrapper officeFactory;
```

Comment:
Factory is not a very good name, as it gives the idea of having a office factory (like the factory design pattern) and I don't think this is what you have in here. Maybe you want to rename the class altogether as `OfficeUtils` / `OfficeManagerUtils`? I know in general is good to group by a higher purpose, to have specific classes., while Utils is way too generic and not that desired. But in your case you simply want to refactor so you can test as much as possible. So Utils sounds more appropriate in here to me
WDYT?